File: javascript/pangram/pangram.js
const alphabet = "abcdefghijklmnopqrstuvwxyz".split("")

function isPangram(text) {
  const chars = text.toLowerCase().replace(/[^a-z]/, "").split("")
  return alphabet.every(letter => chars.includes(letter))
}

function Pangram(text) {
  this.pangram = isPangram(text)
}

Pangram.prototype.isPangram = function () { return this.pangram }

module.exports = Pangram

File: haskell/pangram/src/Pangram.hs
module Pangram (isPangram) where

import Data.Char (isLetter, toLower)
import Data.List (filter)
import Data.List.Unique (sortUniq)

isPangram :: String -> Bool
isPangram text =
  (sortUniq (map toLower (filter isLetter text))) == "abcdefghijklmnopqrstuvwxyz"
